kafka-test/src/test/java/io/specmesh/kafka/ClientsFunctionalDemoTest.java

```diff
@@ -54,20 +55,19 @@
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 class ClientsFunctionalDemoTest extends AbstractContainerTest {
     private static final KafkaApiSpec apiSpec = new KafkaApiSpec(getAPISpecFromResource());
-    private AdminClient adminClient;
-    private SchemaRegistryClient schemaRegistryClient;
+    private final AdminClient adminClient;
+    private final SchemaRegistryClient schemaRegistryClient;
 
-    @BeforeEach
-    public void createAllTheThings() {
-        adminClient = AdminClient.create(getClientProperties());
+    ClientsFunctionalDemoTest() throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
More normal to do this kind of thing in a `setUp` method than in a constructor.
I'd also just `throws Exception` as less noise.

```suggestion
    @BeforeTest
    setUp() throws Exception {
```

On a more general note, it looks like this is bringing up a fresh Kafka/SR for each test, which is surely unnecessary and slows things down.  

Could this not just bring up the docker images once, provision in a `beforeClass` and then run a load of tests? Or am I missing someting?

Reply:
Was using an abstract class so I could add other demo tests with the same containers... but yeah - that could be pulling into a static method in the class to so the same thing